cirq/ops/pauli.py

```diff
@@ -11,29 +11,31 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import abc
+from typing import Any, Union, overload, TYPE_CHECKING
 
-from typing import Optional, Union, overload, TYPE_CHECKING
-
-import numpy as np
-
-from cirq import protocols, value
-from cirq.ops import common_gates
-from cirq.type_workarounds import NotImplementedType
+from cirq import value
+from cirq.ops import common_gates, eigen_gate
 
 if TYPE_CHECKING:
     # pylint: disable=unused-import
     from typing import Tuple
 
 
-@value.value_equality
-class Pauli:
-    """Represents the X, Y, or Z axis of the Bloch sphere."""
+class Pauli(eigen_gate.EigenGate, metaclass=abc.ABCMeta):
```

Comment:
I think we should significantly downgrade this class. Turns all of its methods about pairs of Paulis into private utility methods.

Reply:
Done for `difference()`. The other two already have clients, e.g. in clifford_gate.py and pauli_string.py.

(We could still do this, but it probably makes sense to do so in a separate PR.)

Reply:
Okay, we can push it to a separate PR.